test/cases/ostree-vsphere.sh

```diff
@@ -0,0 +1,884 @@
+#!/bin/bash
+
+# This test scrip covers:
+# 1. build edge-container image, build edge-vsphere image
+# 2. upload image to vsphere datacenter7.0-intel, create edge vm, run test cases against vm.
+# 3. upload image to vsphere datacenter7.0-amd, create edge vm, run test cases against vm.
+# 4. build upgrade edge-container image
+# 5. run rpm-ostree upgrade in all edge vms, then run test cases against upgraded vms.
+# 6. delete all vsphere vms, images
+
+set -euo pipefail
+
+# Provision the software under test.
+/usr/libexec/osbuild-composer-test/provision.sh none
+
+# Get OS data.
+source /etc/os-release
+ARCH=$(uname -m)
+
+source /usr/libexec/tests/osbuild-composer/shared_lib.sh
+
+# Install sshpass
+sudo dnf install -y sshpass
+
+# Install ansible module
+sudo ansible-galaxy collection install community.vmware
```

Comment:
You decided to use govc already. Why not  use govc to create and delete VM instead of ansible playbook.
So the only dependency will be govc.

Reply:
I have tried to govc to do all the upload/create/delete  things, and decided to only use it to upload image because:
1. Use govc to create vm, but failed to use existing hard disk, it always create it's own disk even I specified the existing hard disk. Still trying to fix this, but no progress yet.
2. Use govc to remove vm, this can be done, but considering ansible playbook is more stable and govc may throw error, so still use ansible to remove just like using it to create.

Reply:
-disk=                 Disk path (to use existing) OR size (to create new, e.g. 20GB)
  -disk-datastore=       Datastore for disk file
  -disk.controller=scsi  Disk controller type

Reply:
This is the command I used:
govc vm.create -dc="Datacenter7.0-AMD" -ds="datastore-21" -pool="/Datacenter7.0-AMD/host/Cluster7.0-AMD/Resources" -disk="af432432fdsf8094802480932/af432432fdsf8094802480932.vmdk" -net="VM Network" -net.adapter=vmxnet3 -disk.controller=pvscsi -on=true -c=2 -m=4096 -firmware=efi yi001

I also tried other options, but it will always create a new disk and associate it with vm, and I cannot find the existing disk in vm configuration, the new disk is the only disk associated. And of course it cannot boot with a new disk.
But I found this vm is actually related with existing disk, in an unknown way, because when I delete it, the existing disk and folder in datastore is also deleted.

Do you have any successful command sample?

Reply:
I just saw you later comment with command sample, I will try it.

Reply:
It works, thanks. Removed all ansible code.